geonode/contrib/geosites/models.py

```diff
@@ -98,13 +99,14 @@ def post_delete_profile(instance, sender, **kwargs):
 
 
 # Django doesn't propagate the signals to the parents so we need to add the listeners on the children
-signals.post_save.connect(post_save_resource, sender=Layer)
-signals.post_save.connect(post_save_resource, sender=Map)
-signals.post_save.connect(post_save_resource, sender=Document)
-signals.post_save.connect(post_save_site, sender=Site)
-signals.post_delete.connect(post_delete_resource, sender=Layer)
-signals.post_delete.connect(post_delete_resource, sender=Map)
-signals.post_delete.connect(post_delete_resource, sender=Document)
-signals.post_delete.connect(post_delete_site, sender=Site)
-signals.post_save.connect(post_save_profile, sender=Profile)
-signals.post_delete.connect(post_delete_profile, sender=Profile)
+if 'geonode.contrib.geosites' in settings.INSTALLED_APPS:
```

Comment:
No, this is due to tests failing with the included settings file because the models for geosites aren't created in the database (the default settings doesn't include geosites) but the signals still get registered.